src/action-creators/pull.ts

```diff
@@ -113,4 +114,13 @@ const pull = (contextMap: Index<Context>, { maxDepth }: PullOptions = {}): Thunk
   return hasPending
 }
 
+/**
+ * Calls pullDescendants to fetch and update descendants.
+ * WARNING: Unknown behavior if thoughtsPending takes longer than throttleFlushPending.
+ */
+const pull = (contextMap: Index<Context>, { maxDepth }: PullOptions = {}): Thunk<Promise<boolean>> => async (dispatch, getState) => {
+  const result = await pullDescendants(contextMap, { maxDepth }, dispatch, getState)
+  return result
+}
```

Comment:
Could you explain the purpose of separating `pull` from `pullDescendants`? It looks like you're using `pullDescendants` to remove the thunk, but I'm not sure why.

Reply:
We need to have the ability to load the descendant thoughts `synchronously` before calling `existingThoughtMove` reducer. I noticed that we're using `await dispatch` for the `pull` action in some places but it doesn't seem to wait for the thoughts to be loaded (unless that's the intended behavior, we'd need to update those occurrences as well). 
So I separated that function as async and exported the same.

Reply:
> We need to have the ability to load the descendant thoughts `synchronously` before calling `existingThoughtMove` reducer.

It's impossible to load thoughts synchronously; accessing Firebase or IndexedDB is inherently asynchronous.

> I noticed that we're using `await dispatch` for the `pull` action in some places but it doesn't seem to wait for the thoughts to be loaded (unless that's the intended behavior, we'd need to update those occurrences as well).

The `pull` action creator awaits in several places, so it seems like `await dispatch(pull(...))` would wait for the thoughts to be loaded. If not, that could be a bug.